### PR TITLE
Faster processing of dependencies

### PR DESCRIPTION
**Pull request by knokknok**
In reference to #2318, would the following work?

### CODE REVIEW

**Review by atusy — CHANGES_REQUESTED** [review]
@knokknok

Thank you for the PR.
It seems you are on the right direction.
I added some nit-picking comments about docs and formats.

File: R/cache.R [hunk 1]
```diff
@@ -142,6 +142,7 @@ cache_rx = '_[abcdef0123456789]{32}[.](rdb|rdx|RData)$'
 #' is similar to the effect of the \code{dependson} option. It is supposed to be
 #' used in the first chunk of a document and this chunk must not be cached.
 #' @param path Path to the dependency file.
+#' @param chunk_label The chunk label of the current code chunk.
```
[the remark this answers is not in the document]
Comment (atusy) [reply R/cache.R:145]: I would also describe the default behavior that `NULL` builds dependencies for the all the available chunks.

**Review by yihui — APPROVED** [review]
I've revised the PR according to @atusy's feedback. Thanks!

**Comment by atusy** [comment]
My pleasure!
Looks perfect!!

**Pull request merged**